lib/mantav2-migrate/clicommon.js

```diff
@@ -0,0 +1,298 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * Copyright 2020 Joyent, Inc.
+ */
+
+var fs = require('fs');
+var tty = require('tty');
+
+var assert = require('assert-plus');
+var strsplit = require('strsplit');
+
+function objCopy(obj, target) {
+    if (!target) {
+        target = {};
+    }
+    Object.keys(obj).forEach(function(k) {
+        target[k] = obj[k];
+    });
+    return target;
+}
+
+/**
+ * Prompt a user for a y/n answer.
+ *
+ *      cb('y')        user entered in the affirmative
+ *      cb('n')        user entered in the negative
+ *      cb(false)      user ^C'd
+ */
+function promptYesNo(opts_, cb) {
```

Comment:
Had you considered using a 3rd party module for this? prr uses 'prompt' and it seemed good enough for our uses? Example at https://github.com/joyent/prr/blob/master/lib/prr.js#L654

Reply:
Not really. This is a copied around snippet that I've been using in many projects for a while. A light comparison:

Strikes against my thing here:

- it is copied around code:

```
$ rg -g '!deps' 'function prompt'
...
sdcadm/lib/common.js
302:function promptYesNo(opts_, cb) {

joyent-repos/lib/common.js
257:function promptYesNo(opts_, cb) {
370:function promptEnter(prompt, cb) {

node-triton/lib/common.js
586:function promptYesNo(opts_, cb) {
700:function promptEnter(prompt, cb) {
734:function promptField(field, cb) {
812:function promptPassphraseUnlockKey(opts, cb) {

node-imgadm/lib/common.js
242:function promptYesNo(opts_, cb) {

triton-kbmapi/lib/cli/common.js
49:function promptYesNo(opts_, cb) {

smartos-live/src/img/lib/common.js
243:function promptYesNo(opts_, cb) {
```

- it should probably use node's core "readline" module in some way (either via `prompt`, or the `read` module it uses, or using `readline` directly) after ensuring that works for its various use cases


Strikes against `prompt`:

- it is 2M installed (with its deps)
- it is 800 lines (https://github.com/flatiron/prompt/blob/master/lib/prompt.js)
- its latest published version (1.0.0) is this commit:

```
commit 11f2a2b5123a55c5edd9f11e07f4190e6cf61907 (tag: 1.0.0)
Author: indexzero <charlie.robbins@gmail.com>
Date:   2016-02-10T00:42:58-08:00 (4 years, 2 months ago)

    [dist] Version bump. 1.0.0
```

but there have been 27 other commits to its master branch since then that haven't been published to npm that I see. Not sure what is going on there. Not necessarily *bad*.

- It installs a SIGINT event handler that will call process.exit(1) for me. I don't really want that.

- Also the following strikes against `prompt`, written up in a gist (to include the yesno.js test program I used for comparison): https://gist.github.com/trentm/3bec297455cb1786be3e3264d551ff82